app/src/main/kotlin/compiler/intermediate/Register.kt

```diff
@@ -1,23 +1,25 @@
 package compiler.intermediate
 
 // A class representing virtual cpu register
-class Register {
+class Register(private val asmName: String = "VIRTUAL REGISTER") {
```

Comment:
I think we shouldn't have placeholder name for virtual registers, but just throw an error when one is attempted to be printed.

Reply:
You're right, changed.